### PR TITLE
[BUG] fix SCINetForecaster pretrain test failures via smaller test params

### LINKED ISSUE #10279: [BUG] `SCINetForecaster` failure in `test_predict_series_name_preserved`
`SCINetForecaster` fails `test_predict_series_name_preserved` with a `num_samples` related error. There is no `num_samples` variable in the file itself, so it might be an issue with how dependencies (e.g., `torch`)  are interfaced.

Partial error log:

```
test_predict_series_name_preserved[SCINetForecaster-1]
FAILED: num_samples should be a positive integer value, but got num_samples=0
```

### LINKED ISSUE #10493: [BUG] Pre-training tests are failing for `SCINetForecaster`
**Describe the bug**

Some of the pre-training related tests are failing for `SCINetForecaster`.

```python
from sktime.utils.estimator_checks import check_estimator
from sktime.forecasting.scinet import SCINetForecaster

check_estimator(SCINetForecaster)
```
This gives the following output:
```
FAILED: test_predict_series_name_preserved[SCINetForecaster-1]
FAILED: test_pretrain_clone_preserves_state[SCINetForecaster-0-y:1cols]
FAILED: test_pretrain_clone_preserves_state[SCINetForecaster-0-y:2cols]
FAILED: test_pretrain_clone_preserves_state[SCINetForecaster-1-y:1cols]
FAILED: test_pretrain_clone_preserves_state[SCINetForecaster-1-y:2cols]
FAILED: test_pretrain_fit_predict_workflow[SCINetForecaster-1-y:1cols]
FAILED: test_pretrain_fit_predict_workflow[SCINetForecaster-1-y:2cols]
FAILED: test_pretrain_fitted_params_separation[SCINetForecaster-0-y:1cols]
FAILED: test_pretrain_fitted_params_separation[SCINetForecaster-0-y:2cols]
FAILED: test_pretrain_fitted_params_separation[SCINetForecaster-1-y:1cols]
FAILED: test_pretrain_fitted_params_separation[SCINetForecaster-1-y:2cols]
FAILED: test_pretrain_network_preserved_by_fit[SCINetForecaster-1-y:1cols]
FAILED: test_pretrain_network_preserved_by_fit[SCINetForecaster-1-y:2cols]
FAILED: test_pretrain_not_reset_by_fit[SCINetForecaster-1-y:1cols]
FAILED: test_pretrain_not_reset_by_fit[SCINetForecaster-1-y:2cols]
FAILED: test_pretrain_predict_without_fit[SCINetForecaster-0-y:1cols]
FAILED: test_pretrain_predict_without_fit[SCINetForecaster-0-y:2cols]
FAILED: test_pretrain_predict_without_fit[SCINetForecaster-1-y:1cols]
FAILED: test_pretrain_predict_without_fit[SCINetForecaster-1-y:2cols]
FAILED: test_pretrain_state_transitions[SCINetForecaster-0-y:1cols]
FAILED: test_pretrain_state_transitions[SCINetForecaster-0-y:2cols]
FAILED: test_pretrain_state_transitions[SCINetForecaster-1-y:1cols]
FAILED: test_pretrain_state_transitions[SCINetForecaster-1-y:2cols]
FAILED: test_pretrain_with_hierarchical_data[SCINetForecaster-1-y:1cols]
FAILED: test_pretrain_with_hierarchical_data[SCINetForecaster-1-y:2cols]
```

Also the tests take really long to execute, the test-params should be updated to create a smaller network which runs faster with fewer iterations.

**Expected behavior**

All tests should pass.

**Additional context**

These tests are not identified in the regular CI because they run in VM only when the relevant files are touched.

There is another issue reporting the failing test `test_predict_series_name_preserved` in #10279.

**Versions**
<details>

```
System:
    python: 3.14.4 (main, Apr  8 2026, 04:02:31) [GCC 15.2.0]
executable: /home/geetu/venv/main/bin/python
   machine: Linux-7.0.0-27-generic-x86_64-with-glibc2.43

Python dependencies:
          pip: 25.1.1
       sktime: 1.0.1
 scikit-learn: 1.7.2
  scikit-base: 1.0.1
        numpy: 2.4.4
        scipy: 1.17.1
       pandas: 2.3.3
   matplotlib: 3.10.9
       joblib: 1.5.3
huggingface-hub: 1.16.1
        numba: None
     pmdarima: None
pytorch-forecasting: 1.8.0
   skforecast: None
        skpro: 2.13.0
statsforecast: None
  statsmodels: None
 transformers: 5.8.0.dev0
      tsfresh: None
      tslearn: None
        torch: 2.11.0
   tensorflow: None
```

</details>

### PR DESCRIPTION
**Pull request by Nischal1425**
## Reference Issues/PRs
Fixes #10493
Fixes #10279

## What does this implement/fix?
`get_test_params()` used `seq_len=8/16` with `num_levels=3`, exceeding the timepoint count of the fixed-size fixtures in the pretrain/forecaster test suite (10-15 timepoints). `PyTorchTrainDataset` ended up with zero samples, so `DataLoader` raised `num_samples=0`.

Shrunk params to `seq_len=4, num_levels=1, pred_len=2/3` stays within the smallest fixture (10) and respects `seq_len % 2**num_levels == 0`. Also speeds up the tests, per the issue's second complaint.

Verified: `check_estimator(SCINetForecaster)`, 668/668 pass, 0 failures.

#10289 is a separate open PR adding input validation for the same root cause, but doesn't touch `get_test_params()`, complementary, no conflict.

## Does your contribution introduce a new dependency?
No.

## Did you add any tests for the change?
No, existing `test_pretrain_*` / `test_predict_series_name_preserved` now pass and serve as regression coverage.

### CODE REVIEW

**Review by geetu040 — CHANGES_REQUESTED** [review]
Nice, this fixes the CI.

Quick question: In the CI it takes 1m 32s to run the tests for `SCINetForecaster` which is fine, but running the tests locally take longer. Have you tried to run these tests locally and see how long it takes for you? If this is taking significant time, we should update the test params to speed up the tests.

And I think the configuration through test-params should alone be able to create a smaller network, therefore we should have it.

**Comment by Nischal1425** [comment]
Good question tho, I ran the full `check_estimator(SCINetForecaster)` suite locally (668 tests) three times: 95s, 95s, 99s. That lines up with the 1m32s in CI, so it's not notably slower locally.

And yes, the network size is fully controlled through `get_test_params()` already. `num_levels=1` (down from the default 3) combined with `seq_len=4` is what shrinks the network; no other code changes are needed to configure a smaller model for testing.

**Review by geetu040 — APPROVED** [review]
Thanks. The changes are fine, There is nothing wrong with the implementation, only the test-params needed to be adjusted for test-suits.

A better refactor would be to make `PyTorchTrainDataset` support longer lengths with padding, but I would avoid that in this PR.

CC: @fkiraly for review/merge.

**Comment by Nischal1425** [comment]
@geetu040 Thanks for the review, Agreed padding support in `PyTorchTrainDataset` is a good idea but a separate concern, happy to look into that as a follow-up if there's need for it.
